avocado/core/remote/runner.py

```diff
@@ -45,6 +45,9 @@ class RemoteTestRunner(TestRunner):
     remote_version_re = re.compile(r'^Avocado (\d+)\.(\d+)$',
                                    re.MULTILINE)
 
+    #: remoter connection to the remote machine
+    remote = None
```

Comment:
This is something I dislike.  Class attributes should be used for content that is common across all instances of a class.  It's usually a good idea to use it for "constants" or even values that will be shared by all instances (`self.__class__.remote`).

The `remote` here is clearly an instance attribute (that masks the class level attribute) and should be initialized (`self.remote = None`) at the class `__init__()`.